Gems/Atom/Feature/Common/Assets/Shaders/SkyAtmosphere/SkyRayMarching.azsl

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+ // Based on https://github.com/sebh/UnrealEngineSkyAtmosphere by Sébastien Hillaire
+
+#define ENABLE_ATMOSPHERE_SHADOWS 1
+
+#include <viewsrg.srgi>
+#include <Atom/Features/SrgSemantics.azsli>
+#include <Atom/Features/PostProcessing/FullscreenVertex.azsli>
+#include <Atom/Features/ColorManagement/TransformColor.azsli>
+
+#include <SkyAtmosphereCommon.azsli>
+
+struct RayMarchPixelOutputStruct
+{
+    float4 m_luminance : SV_Target0;
+    float4 m_reflection : SV_Target1;
+};
+
+float3 GetSunLuminance(float3 worldPos, float3 worldDir, float planetRadius)
+{
+    if(o_enableSun)
+    {
+        const float angle = dot(worldDir, PassSrg::m_constants.m_sunDirection);
+        const float sunAngle = cos(0.5 * PassSrg::m_constants.m_sunRadiusFactor * 3.14159 / 180.0);
+        if (angle > sunAngle)
+        {
+            float t = RaySphereClosestHitWS(float3(0.0f, 0.0f, 0.0f), planetRadius, worldPos, worldDir);
+            if (t < 0.0f) // no intersection with surface
+            {
+                const float d = (1.0 - angle) / (1.0 - sunAngle) ;
+				const float falloff = pow(1.0 - d, PassSrg::m_constants.m_sunFalloffFactor);
+				if(PassSrg::m_constants.m_sunFalloffFactor <= 1.0)
+				{
+					// crisp sun disk
+                	return lerp(PassSrg::m_constants.m_sunLimbColor, PassSrg::m_constants.m_sunColor, falloff);
+				}
+				else
+				{
+					// soft sun disk
+                	return lerp(PassSrg::m_constants.m_sunLimbColor, PassSrg::m_constants.m_sunColor, falloff) * falloff;
+				}
+            }
+        }
+    }
+    return 0;
+}
+
+RayMarchPixelOutputStruct RayMarchingPS(VSOutput Input, in uint sampleIndex : SV_SampleIndex)
```

Comment:
nit: add a short explanation what is done here, how and a word about performance / complexity